buildsystem/modules/FindOpusfile.cmake

```diff
@@ -33,11 +40,11 @@ find_library(OPUS_LIBRARY
 mark_as_advanced(OPUS_LIBRARY)
 
 
-# handle the QUIETLY and REQUIRED arguments and set OPUSFILE_FOUND to TRUE if
+# handle the QUIETLY and REQUIRED arguments and set OPUS_FOUND to TRUE if
 # all listed variables are TRUE
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Opus DEFAULT_MSG OPUSFILE_LIBRARY OPUS_LIBRARY OPUS_INCLUDE_DIR)
 
 # export the variables
 set(OPUS_LIBRARIES "${OPUSFILE_LIBRARY}" "${OPUS_LIBRARY}")
-set(OPUS_INCLUDE_DIRS "${OPUS_INCLUDE_DIR}" "${OPUS_INCLUDE_DIR}/opus")
+set(OPUS_INCLUDE_DIRS "${OPUSFILE_INCLUDE_DIR}/opus" "${OPUS_INCLUDE_DIR}/opus")
```

Comment:
Why does this change from `${OPUS_INCLUDE_DIR}` as (1) to **(1)/opus** here?

Reply:
I had problems building openage with nix. It was because opusfile.h and opus.h were in different packages (namely opusfile and libopus), and nix put those in different directories. So I changed this file to look explicitly for both files. That's why I added `"${OPUSFILE_INCLUDE_DIR}/opus"`

As far as I can remember, I thought "${OPUS_INCLUDE_DIR}" was unnecessary/wrong, because the related headerfiles are in "${OPUS_INCLUDE_DIR}"/opus and are included as <opusfile.h> or <opus.h>.
Fact check: 
```
$ ag 'opus[a-z_-]*\.h'
buildsystem/modules/FindOpusfile.cmake
10:#  OPUS_INCLUDE_DIRS   - where to find opus/opus.h, opus/opusfile.h, etc
16:	NAMES opus/opusfile.h
23:	NAMES opus/opus.h
[...]

libopenage/audio/opus_loading.h
7:#include <opus/opusfile.h>

libopenage/audio/opus_in_memory_loader.cpp
[...]
5:#include <opusfile.h>
[...]

libopenage/audio/opus_dynamic_loader.cpp
[...]
5:#include <opusfile.h>

openage/convert/opus/opus.pxd
3:cdef extern from "opus/opus.h":
17:cdef extern from "opus/opus_defines.h":
```
So we have a mixed inclusion style. opus_in_memory_loader.cpp and opus_dynamic_loader.cpp are different from the others. As we are looking in FindOpusfile.cmake for opus/opusfile.h, we should include it everywhere is that.
I propose changing this line to
```
set(OPUS_INCLUDE_DIRS "${OPUSFILE_INCLUDE_DIR}" "${OPUS_INCLUDE_DIR}")
```

Reply:
ah, good catch.

I'd say we should unify the layout to only use `#include <opus/opusfile.h>` and `<opus/opus.h>`, because that's how we did it with all other software packages (hopefully).

For example we used:
```
#include <gperftools/profiler.h>
#include <eigen3/Eigen/Dense>
#include <epoxy/gl.h>
#include <SDL2/SDL.h>
#include <SDL2/SDL_image.h>
...
```

`sdl2` and `sdl2_image` are also separate packages, just like `opus` and `opusfile`.

`opusfile` and `opus` should provide us with their includedirs, and we should add both to the search path in such a way we can write `#include <opus/opusfile.h>` in cpp/h files. Agreed?

Reply:
Yes.
But it doesn't work:
opusfile.h includes <opus_multistream.h>. [Explanation](https://gitlab.xiph.org/xiph/opusfile/commit/9bc622bbe1822584eb8d774700a11b592f462b49)
So looks like we need 
```
set(OPUS_INCLUDE_DIRS "${OPUSFILE_INCLUDE_DIR}" "${OPUS_INCLUDE_DIR}" "${OPUS_INCLUDE_DIR}/opus")
```
Did that

Reply:
hm weird, why do they assume this? :)
anyway, thanks for the fix, that should do it.

Reply:
I guess pkg-config does that, so that you can have several versions installed in different subdirs under include/.